### DISCUSSION TITLE
Error with the Module not found: Error: Can't resolve 'canvas'

### DISCUSSION THREAD

**Discussion by Idletalk** [konvajs/konva#1636] [Q&A]
When I tried Knova, I encountered a problem 【Module not found: Error: Can't resolve 'canvas'】.Then I noticed that the package file would have an "import require$$1 from 'canvas'", and then I went to look for the map file, which showed that "konva/lib/index-node.js" was introduced, and canvas would be introduced in this file.  How should I make a reference to "konva/lib/index.js"?

**Comment by lavrton** [comment]
What is your dev environment? Next.js?

**Reply by watercolord** [reply]
I have the same problem in Quasar (v 1.22, Vue 2), SSR app, the library is imported only on the client.

quasar.config.js:
```js
return {
  boot: [
    {
      server: false,
      path: 'konva',
    }
  ]
}
```

**Reply by lavrton** [reply]
@SmirnovMaxim what will happen if you install `canvas` library manually?

**Reply by watercolord** [reply]
@lavrton, some kind of webpack loader is needed but I don't understand which one.

![image](https://github.com/konvajs/konva/assets/9385067/c4139d1c-2e9d-4cf8-b2bb-76c5fc4ee7ce)

**UPD**: [node-loader](https://github.com/webpack-contrib/node-loader) is needed (I'm using 1.0.3v for webpack 4), the build was successful. However, when opening the canvas page I received this error (when opening another page everything is fine and I can go to the canvas page, but this is very strange behavior):

![image](https://github.com/konvajs/konva/assets/9385067/16cba50e-4e9d-45de-b181-cd7203fa8eae)

**Reply by watercolord** [reply]
So, a few moments later...

Importing konva/lib/index solved my problem

```js
import Konva from 'konva/lib/index'
```

**Comment by lavrton** [comment]
A little note: from `konva@10.0.0` it should load wihtout extra config.
